### PR TITLE
Implement postNewTrigger method(2/3 and 3/3)

### PR DESCRIPTION
**Pull request by yongpingchen**
- Implement `ThingIFAPI#postNewTrigger(ServerCode, Predicate, TriggerOptions)` and `ThingIFAPI#postNewTrigger(ServerCode, Predicate)`
- Add small tests

### CODE REVIEW

File: thingif/src/main/java/com/kii/thingif/ThingIFAPI.java [hunk 1]
```diff
@@ -1088,19 +1088,17 @@ private Trigger postServerCodeNewTrigger(
...
+        JSONObject requestBody = options != null ?
+                JsonUtils.newJson(this.gson.toJson(options)) :
+                new JSONObject();
+        try {
+            requestBody.put("predicate", JsonUtils.newJson(this.gson.toJson(predicate, Predicate.class)));
+            requestBody.put("triggersWhat", TriggersWhat.SERVER_CODE.name());
+            requestBody.put("serverCode", JsonUtils.newJson(this.gson.toJson(serverCode)));
+        } catch (JSONException e) {
+            // Won't happen
+        }
```
Comment (shinsuke-nara) [inline thingif/src/main/java/com/kii/thingif/ThingIFAPI.java]: Like [here](https://github.com/KiiPlatform/thing-if-AndroidSDK/pull/235/files#diff-12e2b8b1411bb1a56fce72fd0a2187a8R1023), I prefer to throw `RuntimeException` even if that situation does not occur.

**Comment by shinsuke-nara** [comment]
@yongpingchen Could you check my comment?

**Comment by yongpingchen** [comment]
@shinsuke-nara fixed.

**Comment by shinsuke-nara** [comment]
@yongpingchen Thanks! LGTM!

**Comment by naoki-akimoto** [comment]
LGTM!

**Pull request merged**